cugraph_nx_demo.ipynb

```diff
@@ -0,0 +1,504 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# How to use `cugraph-nx` to accelerate graph analysis while using networkX "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using the cugraph plugin for NetworkX requires Python 3.9, 3.10, or 3.11, in addition to the latest development version of NetworkX. Follow the instructions from `INSTALL.rst` on the NetworkX repository for instructions on installing the development version of `networkx`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, check which version of networkx you have installed. The cugraph plugin is only on the development version of NetworkX, `networkx-3.2rc0-dev0`. If the below command does not show this version, you will need to follow the instructions from `INSTALL.rst` on the NetworkX repository."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Name: networkx\n",
+      "Version: 3.2rc0.dev0\n",
+      "Summary: Python package for creating and manipulating graphs and networks\n",
+      "Home-page: \n",
+      "Author: \n",
+      "Author-email: Aric Hagberg <hagberg@lanl.gov>\n",
+      "License: \n",
+      "Location: /opt/conda/envs/rapids/lib/python3.10/site-packages\n",
+      "Editable project location: /dylanc/week4/networkx\n",
+      "Requires: \n",
+      "Required-by: cfn-lint, cugraph-nx, mapclassify, osmnx, python-louvain, scikit-image\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip show networkx"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true,
+    "tags": []
+   },
+   "source": [
+    "## FOR REPRODUCIBILITY / DEV WORK / TEMP"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following cells will\n",
+    "- uninstall the current version of networkx \n",
+    "- clone the networkx repository one directory below the current working directory\n",
+    "- upgrade networkx to be the current development version with `pip install`\n",
+    "\n",
+    "You may need to restart the kernel to have the development version of networkX be visible."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
```

Comment:
I included the output of the notebook cells so you, or any other reviewer, could reproduce it. The entire ## FOR REPRODUCIBILITY / DEV WORK / TEMP chunk would be removed, since the cells are highly specific to my own workflow, anyway.

I will include the benchmark cells' outputs and clear the output for the rest. If you can provide me your benchmarks from the larger datasets, I can include them on the notebook as a graphic. It would help highlight performance differences, and it would also be useful for other things too.